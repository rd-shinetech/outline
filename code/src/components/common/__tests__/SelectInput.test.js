import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SelectInput from '../SelectInput';


let props = undefined;


beforeAll(() => {
    props = {
        input: {},
        name: 'category',
        label: 'Category',
        defaultOption: '',
        options: [{ value: 'Josh', text: 'Bloch' }],
        meta: { touched: false, error: {}, warning: {} },
    };

    return props;
});



describe('SelectInput.test.js', () => {
    let wrapper = undefined;

    beforeEach(() => {
        wrapper = shallow(<SelectInput {...props} />);
        return wrapper;
    });

    it('renders without crashing', () => {
        expect(wrapper.length).toEqual(1);
    });


    it('renders as expected', () => {
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });


});
