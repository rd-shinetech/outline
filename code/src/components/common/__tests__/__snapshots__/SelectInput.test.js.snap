exports[`SelectInput.test.js renders as expected 1`] = `
<div
  className="form-group">
  <div
    htmlFor="category">
    Category
  </div>
  <div
    className="field">
    <select
      className="form-control"
      name="category">
      <option />
      <option
        value="Josh">
        Bloch
      </option>
    </select>
  </div>
</div>
`;
